### PR TITLE
[release-4.7] Bug 1939061: Sap license management logs gatherer 4.7

### PR DESCRIPTION
**Pull request by Sergey1011010**
Backport

```
The SAP installation may fail in license management pod, so We need error logs to detect the failure.

Get the filtered error logs as followings.

# oc logs deploy/license-management-l4rvh
Found 2 pods, using pod/license-management-l4rvh-74595f8c9b-flgz9
+ iptables -D PREROUTING -t nat -j VSYSTEM-AGENT-PREROUTING
+ true
+ iptables -F VSYSTEM-AGENT-PREROUTING -t nat
+ true
+ iptables -X VSYSTEM-AGENT-PREROUTING -t nat
+ true
+ iptables -N VSYSTEM-AGENT-PREROUTING -t nat
iptables v1.6.2: can't initialize iptables table `nat': Permission denied
Perhaps iptables or your kernel needs to be upgraded.
```
Tested on SAP cluster with different substrings.

## Categories

- [ ] Bugfix
- [X] Enhancement
- [ ] Backporting
- [ ] Others (CI, Infrastructure, Documentation)

## Sample archive

No changes

## Documentation

- `docs/gathered-data.md`

## Unit Tests

No changes

## Privacy

Yes. There are no sensitive data in the newly collected information.

## References

https://issues.redhat.com/browse/CCXDEV-4172
https://bugzilla.redhat.com/show_bug.cgi?id=1939061
https://access.redhat.com/articles/5100521#license-manager-cannot-be-initialized

### CODE REVIEW

**Comment by tremes** [comment]
I noticed that we don't close the channel in `func GatherSAPConfig(g *Gatherer, c chan<- gatherResult)` can you add it please?

**Comment by Sergey1011010** [comment]
@tremes, added it, thanks. Btw, it would make more sense to close it in the calling function so we don't need to do it manually each time.

File: pkg/gather/clusterconfig/openshift_authentication_logs.go [hunk 1]
```diff
@@ -12,8 +12,19 @@ import (
...
 // Location in archive: config/pod/openshift-authentication/logs/{pod-name}/errors.log
 func GatherOpenshiftAuthenticationLogs(g *Gatherer, c chan<- gatherResult) {
-	messagesToSearch := []string{
-		"AuthenticationError: invalid resource name",
+	defer close(c)
+
+	containersFilter := logContainersFilter{
+		namespace: "openshift-authentication",
+    labelSelector: "app=oauth-openshift",
+	}
```
Comment (tremes) [inline pkg/gather/clusterconfig/openshift_authentication_logs.go:20]: Hopefully last detail - formatting please.

File: pkg/gather/clusterconfig/openshift_authentication_logs.go [hunk 2]
```diff
@@ -24,17 +35,13 @@ func GatherOpenshiftAuthenticationLogs(g *Gatherer, c chan<- gatherResult) {
...
-		messagesToSearch,
-		86400,   // last day
-		1024*64, // maximum 64 kb of logs
+		containersFilter,
+		messagesFilter,
 		"errors",
-		"app=oauth-openshift",
-		false,
-	)
+  )
```
Comment (tremes) [inline pkg/gather/clusterconfig/openshift_authentication_logs.go:44]: Same here I guess.

File: pkg/gather/clusterconfig/openshift_authentication_logs.go (hunk 1 above)
Comment (Sergey1011010) [reply pkg/gather/clusterconfig/openshift_authentication_logs.go:20]: oops. Fixed by `go fmt ./...` :)

**Comment by tremes** [comment]
I went through the code and executed locally on basic cluster (no SAP). Looks good to me.  Thanks 
/lgtm

**Pull request merged**